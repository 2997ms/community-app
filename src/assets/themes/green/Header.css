.tc-communities__header__container {
  background-color: green;
}

.tc-communities__header__menu-link,
.tc-communities__header__menu-link:visited {
  color: #fff;
}

.tc-communities__header__menu-link:hover {
  color: #2886af;
}
