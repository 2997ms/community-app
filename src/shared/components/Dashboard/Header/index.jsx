import React from 'react';
import PT from 'prop-types';

import Handle from 'components/Handle';
import Badge from './Badge';
import './styles.scss';

export default function Header(props) {
  const { title, profile, financials } = props;
  return (
    <div styleName="header-dashboard">
      <div styleName="page-state-header">
        <header>
          <div styleName="page-info">
            <h1>{title}</h1>
          </div>
          <div styleName="info">
            {
              profile && <div styleName="user-metrics">
                <Handle handle={profile.handle} size={24} rating={profile.maxRating.rating} />
                {
                  financials > 0 &&
                  <div styleName="money-earned">
                    <p styleName="number">${financials.toLocaleString()}</p>
                    <p>Earned</p>
                  </div>
                }
              </div>
            }
            <Badge />
          </div>
        </header>
      </div>
    </div>
  );
}

Header.propTypes = {
  title: PT.string,
  profile: PT.shape(),
  financials: PT.number,
};

Header.defaultProps = {
  title: '',
  profile: {
    maxRating: {},
  },
  financials: 0,
};
