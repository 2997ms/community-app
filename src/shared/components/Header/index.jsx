import _ from 'lodash';
import React, { useState } from 'react';
import PT from 'prop-types';
import { config } from 'topcoder-react-utils';
import { TopNav, LoginNav } from 'navigation-component';
import Logo from 'assets/images/tc-logo.svg';

const Header = ({ profile }) => {
  const [activeLevel1Id, setActiveLevel1Id] = useState();

  const handleChangeLevel1Id = (menuId) => {
    setActiveLevel1Id(menuId);
  };

  const handleSwitchMenu = () => {
    setActiveLevel1Id(config.HEADER_MENU[0].id);
  };

  let normalizedProfile = profile && _.clone(profile);
  if (profile) {
    normalizedProfile.photoURL = _.has(profile, 'photoURL') ? `${config.CDN.PUBLIC}/avatar/${
      encodeURIComponent(normalizedProfile.photoURL)}?size=32` : '';
  } else {
    normalizedProfile = null;
  }

  return (
    <div>
      <TopNav
        menu={config.HEADER_MENU}
        rightMenu={(
          <LoginNav
            loggedIn={!_.isEmpty(profile)}
            notificationButtonState="none"
            notifications={[]}
            accountMenu={config.ACCOUNT_MENU}
            switchText={config.ACCOUNT_MENU_SWITCH_TEXT}
            onSwitch={handleSwitchMenu}
            showNotification={false}
            profile={normalizedProfile}
            authURLs={config.HEADER_AUTH_URLS}
          />
        )}
        logo={<Logo />}
        theme={config.HEADER_MENU_THEME}
        currentLevel1Id={activeLevel1Id}
        onChangeLevel1Id={handleChangeLevel1Id}
      />
    </div>
  );
};

Header.defaultProps = {
  profile: null,
};

Header.propTypes = {
  profile: PT.shape({
    photoURL: PT.string,
  }),
};

export default Header;
