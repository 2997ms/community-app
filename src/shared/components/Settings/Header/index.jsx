/**
 * Child component of Settings/Header renders summary of number of active challenges.
 * Also renders special badges based on achievements data.
*/
import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';
import TabBar from '../TabBar';
import Badge, { MAP, XL_MAP } from '../../Dashboard/Header/Badge';

import './styles.scss';

export default function Header(props) {
  const {
    profileState,
    showXlBadge,
    xlBadge,
    activeChallengesCount,
    settingsTab
  } = props;
  const achievements = profileState.achievements || [];
  const badges = achievements.filter(x => MAP[x.description]);
  const hasBadges = badges && badges.length;
  const selectTab = (tab) => {
    console.log("Settings tab", tab);
    var sTab;
    switch (tab){
      case 'profile':{
        sTab= 'basicinfo';
        break;
      }
      case 'tools':{
        sTab= 'devices';
        break;
      }
      case 'preferences':{
        sTab= 'email';
        break;
      }
    }
    const tab1= tab+ "/"+sTab;
    console.log("tab1", tab1);
    const tab2= tab1.split("/")[0];
    const tab3= tab1.split("/")[1];
    console.log("tab2", tab2);
    console.log("tab3", tab3);
    
    props.selectTab(tab1);
    
    props.history.push(`/settings/${tab}`);
  };
  return (
    <div styleName="page-state-header">
      <header>
        <div styleName="page-info">
          <h1>Settings</h1>
        </div>
        <TabBar
          settingsTab= {settingsTab}
          selectTab= {selectTab}
        />
        {/* <div styleName="info">
          <div styleName="badgeItem">
            {
              _.map(badges, ({ description }) => (
                <Badge
                  badge={MAP[description]}
                  key={description}
                  showXl={show => showXlBadge(show && description)}
                  title={description}
                  xlBadge={xlBadge === description && XL_MAP[xlBadge]}
                />
              ))
            }
          </div>
          <div styleName="separator" className={hasBadges ? '' : 'hidden'} />
          <div styleName="item">
            <div styleName="value">
              <p>{_.isNumber(activeChallengesCount) && activeChallengesCount >= 0 ? activeChallengesCount : ''}</p>
            </div>
            <div styleName="title">
              <p>Active Challenges</p>
            </div>
          </div>
        </div> */}
      </header>
    </div>
  );
}

Header.defaultProps = {
  xlBadge: null,
  activeChallengesCount: -1,
};

Header.propTypes = {
  profileState: PT.shape().isRequired,
  showXlBadge: PT.func.isRequired,
  xlBadge: PT.string,
  activeChallengesCount: PT.number,
};

