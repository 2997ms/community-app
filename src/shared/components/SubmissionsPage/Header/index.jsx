import PT from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import './styles.scss';

/**
 * Header shown on Submissions Page
 */
const Header = ({
  challengeId,
  challengesUrl,
  title,
}) => (
  <div styleName="header">
    <Link to={`${challengesUrl}/${challengeId}`}>
      <span>&#x2039;</span>
      <p>Back to challenge</p>
    </Link>
    <h2>{title}</h2>
  </div>
);

Header.propTypes = {
  challengeId: PT.number.isRequired,
  challengesUrl: PT.string.isRequired,
  title: PT.string.isRequired,
};

export default Header;
