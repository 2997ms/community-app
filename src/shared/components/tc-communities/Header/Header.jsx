/**
 * The Header component for communities
 *
 * Along with css modules it has global css classes
 * so the whole design can be overridden by an additional css file
 */

import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';
import { Link, NavLink } from 'react-router-dom';
import './Header.scss';

export default function Header(props) {
  const { logos, menuItems, communityId, cssUrl, isMobileOpen, onMobileToggleClick } = props;

  return (
    <div>
      {cssUrl && <link rel="stylesheet" type="text/css" href={cssUrl} />}
      <header styleName="container" className="tc-communities__header__container">
        <div styleName="header" className="tc-communities__header__header">
          <div styleName="logos" className="tc-communities__header__logos">
            {_.map(logos, (logoUrl, index) =>
              (menuItems.length ? (
                <Link
                  key={index}
                  to={`/community/${communityId}/${menuItems[0].url}`}
                  styleName="logo"
                  className="tc-communities__header__logo"
                >
                  <img src={logoUrl} alt="Community logo" />
                </Link>
              ) : (
                <span
                  key={index}
                  styleName="logo"
                  className="tc-communities__header__logo"
                >
                  <img src={logoUrl} alt="Community logo" />
                </span>
              )),
            )}
          </div>
          <button
            styleName="mobile-toggle"
            className="tc-communities__header__mobile-toggle"
            onClick={onMobileToggleClick}
          >
            <span>Toggle navigation</span>
            <i /><i /><i />
          </button>
        </div>
        <div
          styleName={`menu-wrap${isMobileOpen ? ' open' : ''}`}
          className={`tc-communities__header__menu-wrap${isMobileOpen ? ' tc-communities__header__open' : ''}`}
        >
          <ul styleName="menu" className="tc-communities__header__menu">
            {_.map(menuItems, item => (
              <li
                styleName="menu-item"
                className="tc-communities__header__menu-item"
                key={item.url}
              >
                <NavLink
                  styleName="menu-link"
                  className="tc-communities__header__menu-link"
                  activeClassName="menu-link_active tc-communities__header__menu-link_active"
                  to={`/community/${communityId}/${item.url}`}
                >
                  {item.title}
                </NavLink>
              </li>
            ))}
          </ul>
        </div>
      </header>
    </div>
  );
}

Header.defaultProps = {
  menuItems: [],
  logos: [],
  isMobileOpen: false,
  cssUrl: null,
  communityId: null,
};

Header.propTypes = {
  menuItems: PT.arrayOf(PT.shape({
    title: PT.string.isRequired,
    url: PT.string.isRequired,
  })),
  logos: PT.arrayOf(PT.string),
  isMobileOpen: PT.bool,
  cssUrl: PT.string,
  onMobileToggleClick: PT.func.isRequired,
  communityId: PT.string,
};
