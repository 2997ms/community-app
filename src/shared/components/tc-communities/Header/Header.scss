@import '~styles/tc-styles';

.avatar {
  display: inline-block;
  margin: 12px 6px 8px 12px;
}

.avatar-mobile {
  float: right;
  margin: 19px 0 8px 12px;
}

.user-menu {
  cursor: pointer;
  display: inline-block;
  height: 100%;
  margin-bottom: 6px;
  margin-left: 24px;
  vertical-align: bottom;
}

@media (max-width: #{$md-width}) {
  .user-menu {
    display: none;
  }
}

@media (min-width: #{$md-width}) {
  .avatar-mobile {
    display: none;
  }
}

.user-menu-handle {
  // border-bottom: 2px solid $tc-dark-blue-90;
  display: inline-block;
  font-size: 15px;
  height: 54px;
  padding: 20px 0;
  vertical-align: bottom;
}

.container {
  background-color: #fff;
  border-bottom: 1px solid #e7e7e7;

  @include roboto;

  a {
    cursor: pointer;
  }

  &::after {
    clear: both;
    content: '';
    display: table;
    width: 100%;
  }
}

.logo {
  & + .logo {
    margin-left: 20px;
  }

  > img {
    display: block;
    max-height: 40px;
  }

  @include xxs-to-sm {
    & + .logo {
      margin-left: 10px;
    }
  }
}

.logos {
  align-items: center;
  display: flex;
  float: left;
  height: 100%;
}

.header {
  float: none;
  height: 66px;
  padding: 0 20px;

  &::after {
    clear: both;
    content: '';
    display: table;
    width: 100%;
  }

  @media (min-width: #{$md-width}) {
    float: left;
  }
}

.menu-wrap {
  max-height: 0;
  overflow: hidden;
  transition: max-height 0.3s ease-out;

  &.open {
    max-height: 1000px;
    transition-timing-function: ease-in;
  }

  @media (min-width: #{$md-width}) {
    overflow: visible;
  }
}

.menu {
  border-top: 1px solid #e7e7e7;
  padding: 0 20px;
  text-align: right;
}

.menu-item {
  display: block;
  text-align: left;

  @media (min-width: #{$md-width}) {
    display: inline-block;
  }
}

.menu-link {
  display: block;
  line-height: 66px;
  padding: 0 20px;

  &:hover {
    color: #2886af;
  }

  @media (min-width: #{$md-width}) {
    display: inline-block;
  }
}

/*
  This is a hack for server-rendering

  As we have to change active link style dynamically we have import styles to some variable like
  import styles from './Header.scss' and after use it like className={styles['menu-link_active']}
  But so far on the server when importing styles it returns empty object styles == {}
  So when when we try to use styles['menu-link_active'] we get undefined instead of real class name
  It breaks server rendering.

  To temporary fix it, we create a global style instead so inside Header component we can use
  a constant string instead of variable.

  TODO: when importing styles on server will be fixed to return an object with style list
        the :global has to be removed and inside Header component string "menu-link_active"
        must be replaced with styles['menu-link_active']
 */
:global {
  .menu-link_active,
  .menu-link_active:visited {
    color: #2886af;
  }
}

.mobile-toggle {
  background-color: transparent;
  background-image: none;
  border: 1px solid #ddd;
  border-radius: 4px;
  display: block;
  float: right;
  margin-top: 18px;
  outline: none;
  padding: 9px 10px;
  position: relative;

  &:focus {
    background-color: #ddd;
  }

  > span {
    position: absolute;
    width: 1px;
    height: 1px;
    padding: 0;
    margin: -1px;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    border: 0;
  }

  > i {
    background-color: #888;
    border-radius: 1px;
    display: block;
    height: 2px;
    width: 22px;

    & + i {
      margin-top: 4px;
    }
  }

  @media (min-width: #{$md-width}) {
    display: none;
  }
}
