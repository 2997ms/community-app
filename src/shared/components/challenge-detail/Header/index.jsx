/* eslint jsx-a11y/no-static-element-interactions:0 */
/**
 * Challenge header component.
 * This component renders all other child components part of the header.
 * Any data massaging needed for a child view should be done here.
 */

import config from 'utils/config';
import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';
import moment from 'moment';
import { DangerButton, PrimaryButton } from 'components/buttons';
import { ThemeProvider } from 'react-css-themr';

import ArrowUp from '../../../../assets/images/icon-arrow-up.svg';
import ArrowDown from '../../../../assets/images/icon-arrow-down.svg';
import themeFactory from '../themeFactory';

import Prizes from './Prizes';
import ChallengeTags from './ChallengeTags';
import DeadlineCards from './DeadlineCards';
import ChallengeViewSelector from './ChallengeViewSelector';
import style from './style.scss';

export default function ChallengeHeader(props) {
  const {
    challenge: {
      id: challengeId,
      name,
      track,
      subTrack,
      events,
      technologies,
      platforms,
      prizes,
      numberOfCheckpointsPrizes,
      topCheckPointPrize,
      reliabilityBonus,
      userDetails,
      currentPhases,
      registrationEndDate,
      submissionEndDate,
      numRegistrants,
      numSubmissions,
      allPhases,
    },
    registering,
    registerForChallenge,
    unregisterFromChallenge,
    unregistering,
  } = props;

  const theme = themeFactory(track);

  const trackLower = track ? track.toLowerCase() : 'design';
  const stylizedSubTrack = (subTrack || '').replace('_', ' ')
    .replace(/\w\S*/g, txt => _.capitalize(txt));
  const subTrackStyle = `${trackLower}-accent-background`;
  const eventStyle = `${trackLower}-accent-color`;
  const eventNames = (events || []).map((event => (event.eventName || '').toUpperCase()));
  const miscTags = _.union((technologies || '').split(', '), platforms.split(', '));

  const tagFilterString = '/challenges?filter[tags][0]=';

  let bonusType = '';
  if (numberOfCheckpointsPrizes && topCheckPointPrize) {
    bonusType = 'Bonus';
  } else if (reliabilityBonus) {
    bonusType = 'Reliability Bonus';
  }
  const hasRegistered = userDetails && userDetails.roles && userDetails.roles.includes('Submitter');
  const registrationEnded = new Date(registrationEndDate).getTime() < Date.now();
  const submissionEnded = new Date(submissionEndDate).getTime() < Date.now();
  const hasSubmissions = userDetails && userDetails.hasUserSubmittedForReview;
  const nextDeadline = currentPhases && currentPhases.length > 0 && currentPhases[0].phaseType;
  const deadlineEnd = currentPhases && currentPhases.length > 0 ?
    new Date(currentPhases[0].scheduledEndTime).getTime() : Date.now();
  const currentTime = Date.now();
  const timeDiff = deadlineEnd > currentTime ? deadlineEnd - currentTime : 0;
  const duration = moment.duration(timeDiff);
  const timeLeft = `${duration.days()}d ${duration.hours()}:${duration.minutes()}h`;
  let relevantPhases = [];

  if (props.showDeadlineDetail) {
    relevantPhases = (allPhases || []).filter((phase) => {
      const phaseLowerCase = phase.phaseType.toLowerCase();
      if (phaseLowerCase.includes('screening') || phaseLowerCase.includes('specification')) {
        return false;
      }
      if (phaseLowerCase.includes('registration') || phaseLowerCase.includes('checkpoint') ||
          phaseLowerCase.includes('submission') || phaseLowerCase.includes('approval') ||
          phaseLowerCase.includes('review')) {
        return true;
      }
      return false;
    });

    relevantPhases.sort((a, b) => {
      if (a.phaseType.toLowerCase().includes('registration')
      || b.phaseType.toLowerCase().includes('approval')) {
        return -1;
      }
      if (b.phaseType.toLowerCase().includes('registration')
      || a.phaseType.toLowerCase().includes('approval')) {
        return 1;
      }
      return (new Date(a.actualEndTime || a.scheduledEndTime)).getTime() -
      (new Date(b.actualEndTime || b.scheduledEndTime)).getTime();
    });
  }

  return (
    <ThemeProvider theme={theme} >
      <div styleName="challenge-outer-container">
        <div styleName="important-detail">
          <h1 styleName="challenge-header">{name}</h1>
          <ChallengeTags
            subTrack={stylizedSubTrack}
            events={eventNames}
            technPlatforms={miscTags}
            subTrackStyle={subTrackStyle}
            eventStyle={eventStyle}
            tagFilterString={tagFilterString}
          />
          <div styleName="prizes-ops-container">
            <div styleName="prizes-outer-container">
              <h3 styleName="prizes-title">PRIZES</h3>
              <Prizes prizes={prizes && prizes.length ? prizes : [0]} />
              {
                bonusType &&
                <div id={`bonus-${trackLower}`} styleName="bonus-div">
                  {
                    bonusType === 'Bonus' ?
                      <p styleName="bonus-text">
                        <span styleName={`bonus-highlight ${trackLower}-accent-color`}>
                          BONUS: {numberOfCheckpointsPrizes} </span>CHECKPOINTS AWARDED
                          WORTH <span styleName={`bonus-highlight ${trackLower}-accent-color`}>${topCheckPointPrize} </span>EACH
                      </p> :
                      <p styleName="bonus-text">
                        <span styleName={`bonus-highlight ${trackLower}-accent-color`}>
                          RELIABILITY BONUS: {reliabilityBonus}
                        </span>
                      </p>
                  }
                </div>
              }
            </div>
            <div styleName="challenge-ops-container">
              {hasRegistered ? (
                <DangerButton
                  disabled={unregistering || registrationEnded}
                  onClick={unregisterFromChallenge}
                  theme={{ button: style.challengeAction }}
                >Unregister</DangerButton>
              ) : (
                <PrimaryButton
                  disabled={registering || registrationEnded}
                  onClick={registerForChallenge}
                  theme={{ button: style.challengeAction }}
                >Register</PrimaryButton>
              )}
              <PrimaryButton
                disabled={!hasRegistered || unregistering || submissionEnded}
                theme={{ button: style.challengeAction }}
                to={trackLower === 'design' ?
                  `${config.URL.BASE}/challenges/${challengeId}/submit/file` :
                  `${config.URL.BASE}/challenge-details/${challengeId}/submit/?type=develop`
                }
              >Submit</PrimaryButton>
              <PrimaryButton
                disabled={!hasRegistered || unregistering || !hasSubmissions}
                theme={{ button: style.challengeAction }}
                to={`/challenges/${challengeId}/my-submissions`}
              >View Submissions</PrimaryButton>
            </div>
          </div>
          <div styleName="deadlines-view">
            <div styleName="deadlines-overview">
              <div styleName="deadlines-overview-text">
                <div styleName="next-deadline">
                  Next Deadline: <span styleName="deadline-highlighted">{nextDeadline || '-'}</span>
                </div>
                <div styleName="current-phase">
                  <span styleName="deadline-highlighted">
                    {timeLeft}
                  </span> until current deadline ends
                </div>
              </div>
              <a onClick={props.onToggleDeadlines} styleName="deadlines-collapser">
                {props.showDeadlineDetail ?
                  <span styleName="collapse-text">Hide Deadlines <ArrowDown /></span>
                  : <span styleName="collapse-text">View All Deadlines <ArrowUp /></span>
                }
              </a>
            </div>
            {
              props.showDeadlineDetail &&
              <DeadlineCards relevantPhases={relevantPhases} />
            }
          </div>
          <ChallengeViewSelector
            onSelectorClicked={props.onSelectorClicked}
            trackLower={trackLower}
            selectedView={props.selectedView}
            numRegistrants={numRegistrants}
            numSubmissions={numSubmissions}
          />
        </div>
      </div>
    </ThemeProvider>
  );
}

ChallengeHeader.propTypes = {
  onSelectorClicked: PT.func.isRequired,
  selectedView: PT.string.isRequired,
  challenge: PT.shape({
    id: PT.number.isRequired,
  }).isRequired,
  registerForChallenge: PT.func.isRequired,
  showDeadlineDetail: PT.bool.isRequired,
  onToggleDeadlines: PT.func.isRequired,
  registering: PT.bool.isRequired,
  unregisterFromChallenge: PT.func.isRequired,
  unregistering: PT.bool.isRequired,
};
