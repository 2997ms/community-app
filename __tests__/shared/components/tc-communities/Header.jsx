import React from 'react';
import Rnd from 'react-test-renderer/shallow';
import Header from 'components/tc-communities/Header';
import TU from 'react-dom/test-utils';

const mockOnMobileToggle = jest.fn();

const rnd = new Rnd();

test('Snapshot match', () => {
  rnd.render((
    <Header
      onMobileToggleClick={mockOnMobileToggle}
      communityId="someId"
    />
  ));
  expect(rnd.getRenderOutput()).toMatchSnapshot();

  rnd.render((
    <Header
      onMobileToggleClick={mockOnMobileToggle}
      communityId="someId"
      logoUrl="some/logo/url"
      menuItems={[
        { title: 'Menu Item 1', url: '/menu/item/1' },
        { title: 'Menu Item 2', url: '/menu/item/2' },
        { title: 'Menu Item 3', url: '/menu/item/3' },
      ]}
      cssUrl="some/css/url"
      isMobileOpen
    />
  ));
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});

class Wrapper extends React.Component {
  componentDidMount() {}
  render() {
    return <div><Header {...this.props} /></div>;
  }
}

const page = TU.renderIntoDocument((
  <Wrapper
    onMobileToggleClick={mockOnMobileToggle}
    communityId="someId"
  />
));

describe('Toggle mobile menu', () => {
  beforeEach(() => jest.clearAllMocks());

  test('onMobileToggle', () => {
    const btn = TU.findAllInRenderedTree(page, item =>
      item && item.className && item.className.match(/mobile-toggle/));
    expect(btn.length).toBe(1);
    TU.Simulate.click(btn[0]);
    expect(mockOnMobileToggle).toHaveBeenCalled();
  });
});
