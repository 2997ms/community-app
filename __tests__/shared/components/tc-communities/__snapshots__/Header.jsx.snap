// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot match 1`] = `
<div>
  <header
    className="tc-communities__header__container src-shared-components-tc-communities-Header-___Header__container___YEWV6"
  >
    <div
      className="tc-communities__header__header src-shared-components-tc-communities-Header-___Header__header___1vQf1"
    >
      <div
        className="tc-communities__header__logos src-shared-components-tc-communities-Header-___Header__logos___1u4yY"
      >
        
      </div>
      <button
        className="tc-communities__header__mobile-toggle src-shared-components-tc-communities-Header-___Header__mobile-toggle___JaNvG"
        onClick={[Function]}
      >
        <span>
          Toggle navigation
        </span>
        <i />
        <i />
        <i />
      </button>
    </div>
    <div
      className="tc-communities__header__menu-wrap src-shared-components-tc-communities-Header-___Header__menu-wrap___4JXX4"
    >
      <ul
        className="tc-communities__header__menu src-shared-components-tc-communities-Header-___Header__menu___2U-Hb"
      >
        
      </ul>
    </div>
  </header>
</div>
`;

exports[`Snapshot match 2`] = `
<div>
  <link
    href="some/css/url"
    rel="stylesheet"
    type="text/css"
  />
  <header
    className="tc-communities__header__container src-shared-components-tc-communities-Header-___Header__container___YEWV6"
  >
    <div
      className="tc-communities__header__header src-shared-components-tc-communities-Header-___Header__header___1vQf1"
    >
      <div
        className="tc-communities__header__logos src-shared-components-tc-communities-Header-___Header__logos___1u4yY"
      >
        <Link
          className="tc-communities__header__logo src-shared-components-tc-communities-Header-___Header__logo___IYxzO"
          replace={false}
          to="/community/someId/pageId1"
        >
          <img
            alt="Community logo"
            src="some/logo/url"
          />
        </Link>
      </div>
      <button
        className="tc-communities__header__mobile-toggle src-shared-components-tc-communities-Header-___Header__mobile-toggle___JaNvG"
        onClick={[Function]}
      >
        <span>
          Toggle navigation
        </span>
        <i />
        <i />
        <i />
      </button>
    </div>
    <div
      className="tc-communities__header__menu-wrap tc-communities__header__open src-shared-components-tc-communities-Header-___Header__menu-wrap___4JXX4 src-shared-components-tc-communities-Header-___Header__open___3Qq9v"
    >
      <ul
        className="tc-communities__header__menu src-shared-components-tc-communities-Header-___Header__menu___2U-Hb"
      >
        <li
          className="tc-communities__header__menu-item src-shared-components-tc-communities-Header-___Header__menu-item___3LG8-"
        >
          <NavLink
            activeClassName="menu-link_active tc-communities__header__menu-link_active"
            className="tc-communities__header__menu-link src-shared-components-tc-communities-Header-___Header__menu-link___Dm7_k"
            to="/community/someId/pageId1"
          >
            Menu Item 1
          </NavLink>
        </li>
        <li
          className="tc-communities__header__menu-item src-shared-components-tc-communities-Header-___Header__menu-item___3LG8-"
        >
          <NavLink
            activeClassName="menu-link_active tc-communities__header__menu-link_active"
            className="tc-communities__header__menu-link src-shared-components-tc-communities-Header-___Header__menu-link___Dm7_k"
            to="/community/someId/pageId2"
          >
            Menu Item 2
          </NavLink>
        </li>
        <li
          className="tc-communities__header__menu-item src-shared-components-tc-communities-Header-___Header__menu-item___3LG8-"
        >
          <NavLink
            activeClassName="menu-link_active tc-communities__header__menu-link_active"
            className="tc-communities__header__menu-link src-shared-components-tc-communities-Header-___Header__menu-link___Dm7_k"
            to="/community/someId/pageId3"
          >
            Menu Item 3
          </NavLink>
        </li>
      </ul>
    </div>
  </header>
</div>
`;
